### ISSUE TITLE
BUG: inference of unit64s are wrong

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas#16109]
```
In [1]: pd.api.types.infer_dtype([2*63, -1])
Out[1]: 'integer'

In [3]: pd.api.types.infer_dtype([2*63])
Out[3]: 'integer'
```
[1] should be 'mixed' (inference which means object)
[2] should be 'uinteger'

note this will probably break some things in other parts of the suite as we sometimes infer to figure out the dtype and act on it, so 'uinteger' would then be unhandled.

e.g. in Index creation and https://github.com/pandas-dev/pandas/pull/16108/files

**Comment by jreback** [comment]
cc @gfyoung

**Comment by gfyoung** [comment]
@jreback : This was a deliberate choice to consider `integer` to include ALL integer data types (more like a Python `int` in this sense).  Afterwards, we handled `uint` by checking if all were `>= 0`.  So wouldn't necessarily call this a BUG (more like an API change?)

**Comment by jreback** [comment]
yes this is more of a change. I think we *should* differentiate these.
